File: KLine-Chart-OC/KLine-Chart-OC/KLineChart/View/KLineInfoView.m
//
//  KLineInfoView.m
//  KLine-Chart-OC
//
//  Created by 何俊松 on 2020/3/10.
//  Copyright © 2020 hjs. All rights reserved.
//

#import "KLineInfoView.h"
#import "ChartStyle.h"

@interface KLineInfoView()
@property (weak, nonatomic) IBOutlet UILabel *timeLable;

@property (weak, nonatomic) IBOutlet UILabel *openLable;
@property (weak, nonatomic) IBOutlet UILabel *highLable;
@property (weak, nonatomic) IBOutlet UILabel *lowLabel;
@property (weak, nonatomic) IBOutlet UILabel *clsoeLabel;
@property (weak, nonatomic) IBOutlet UILabel *IncreaseLabel;
@property (weak, nonatomic) IBOutlet UILabel *amplitudeLabel;
@property (weak, nonatomic) IBOutlet UILabel *amountLable;

@end

@implementation KLineInfoView

+(instancetype)lineInfoView {
    KLineInfoView *view = [[NSBundle mainBundle] loadNibNamed:@"KLineInfoView" owner:self options:nil].lastObject;
    view.frame = CGRectMake(0, 0, 120, 145);
    return view;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    self.backgroundColor = ChartColors_bgColor;
    self.layer.borderWidth = 1;
    self.layer.borderColor = ChartColors_gridColor.CGColor;
}

- (void)setModel:(KLineModel *)model {
    _model = model;
//    _timeLable.text =
    NSDate *date = [NSDate dateWithTimeIntervalSince1970:model.id];
    NSDateFormatter *formater = [[NSDateFormatter alloc] init];
    formater.dateFormat = @"yy-MM-dd HH:mm";
    _timeLable.text = [formater stringFromDate:date];
    
    _openLable.text = [NSString stringWithFormat:@"%.2f",model.open];
     _highLable.text = [NSString stringWithFormat:@"%.2f",model.high];
     _lowLabel.text = [NSString stringWithFormat:@"%.2f",model.low];
     _clsoeLabel.text = [NSString stringWithFormat:@"%.2f",model.close];
    CGFloat upDown = model.close - model.open;
    NSString *symbol = @"-";
    if(upDown > 0) {
        symbol = @"+";
        self.IncreaseLabel.textColor = ChartColors_upColor;
        self.amplitudeLabel.textColor = ChartColors_upColor;
    } else {
        self.IncreaseLabel.textColor = ChartColors_dnColor;
        self.amplitudeLabel.textColor = ChartColors_dnColor;
    }
    CGFloat upDownPercent = upDown / model.open * 100;
    _IncreaseLabel.text = [NSString stringWithFormat:@"%@%.2f",symbol,ABS(upDown)];
    _amplitudeLabel.text = [NSString stringWithFormat:@"%@%.2f%%",symbol,ABS(upDownPercent)];
    _amountLable.text = [NSString stringWithFormat:@"%.2f",model.vol];
    
}

@end

File: KLine-Chart/KLine-Chart/KLineChart/View/KLineInfoView.swift
//
//  KLineInfoView.swift
//  KLine-Chart
//
//  Created by 何俊松 on 2020/3/2.
//  Copyright © 2020 hjs. All rights reserved.
//

import UIKit

class KLineInfoView: UIView {
    @IBOutlet var timeLable: UILabel!

    @IBOutlet var openLable: UILabel!

    @IBOutlet var highLable: UILabel!

    @IBOutlet var lowLabel: UILabel!

    @IBOutlet var clsoeLabel: UILabel!

    @IBOutlet var IncreaseLabel: UILabel!

    @IBOutlet var amplitudeLabel: UILabel!

    @IBOutlet var amountLable: UILabel!

    var model: KLineModel? {
        didSet {
            guard let _model = model else {
                return
            }
            timeLable.text = calculateDateText(timestamp: _model.id, dateFormat: "yy-MM-dd HH:mm")
            openLable.text = String(format: "%.2f", _model.open)
            highLable.text = String(format: "%.2f", _model.high)
            lowLabel.text = String(format: "%.2f", _model.low)
            clsoeLabel.text = String(format: "%.2f", _model.close)
            let upDown = _model.close - _model.open
            var symbol = "-"
            if upDown > 0 {
                symbol = "+"
                self.IncreaseLabel.textColor = ChartColors.upColor
                self.amplitudeLabel.textColor = ChartColors.upColor
            } else {
                self.IncreaseLabel.textColor = ChartColors.dnColor
                self.amplitudeLabel.textColor = ChartColors.dnColor
            }
            let upDownPercent = upDown / _model.open * 100
            IncreaseLabel.text = symbol + String(format: "%.2f", abs(upDown))
            amplitudeLabel.text = symbol + String(format: "%.2f", abs(upDownPercent)) + "%"
            amountLable.text = String(format: "%.2f", _model.vol)
        }
    }

    override func awakeFromNib() {
        super.awakeFromNib()
        backgroundColor = ChartColors.bgColor
        layer.borderWidth = 1
        layer.borderColor = ChartColors.gridColor.cgColor
    }

    static func lineInfoView() -> KLineInfoView {
        guard let view = Bundle.main.loadNibNamed("KLineInfoView", owner: self, options: nil)?.last as? KLineInfoView else {
            fatalError()
        }
        view.frame = CGRect(x: 0, y: 0, width: 120, height: 145)
        return view
    }
}
